src/Strategy/Swift/PackageSwift.hs

```diff
@@ -0,0 +1,248 @@
+
+parsePackageSwiftFile :: Parser SwiftPackage
+parsePackageSwiftFile = do
+  -- Package.swift must specify version for swift tools
+  -- https://github.com/apple/swift-package-manager/blob/main/Documentation/PackageDescription.md#about-the-swift-tools-version
+  swiftToolVersion <- parseSwiftToolVersion
+  SwiftPackage swiftToolVersion <$> parsePackageDependencies
+
+-- | Analysis
+-- *
+analyzePackageSwift :: (Has ReadFS sig m, Has Diagnostics sig m) => Path Abs File -> m (Graphing.Graphing Dependency, GraphBreadth)
+analyzePackageSwift manifestFile = do
+  manifestContent <- context "Identifying dependencies in Package.swift" $ readContentsParser parsePackageSwiftFile manifestFile
+  graph <- context "Building dependency graph" $ pure $ buildGraph manifestContent
+  pure (graph, Partial)
+
+-- | Graph Building
+-- *
+buildGraph :: SwiftPackage -> Graphing.Graphing Dependency
+buildGraph pkg = induceJust $ directs (map toDependency $ packageDependencies pkg)
+
+toDependency :: SwiftPackageDep -> Maybe Dependency
+toDependency (GitSource pkgDep) =
+  Just $
+    Dependency
+      { dependencyType = depType
+      , dependencyName = srcOf pkgDep
+      , dependencyVersion =
+          CEq
+            <$> asum
+              [ branchOf pkgDep
+              , revisionOf pkgDep
+              , exactOf pkgDep
+              , toFromExpression <$> fromOf pkgDep
+              , toUpToNextMajorExpression <$> upToNextMajorOf pkgDep
+              , toUpToNextMinorExpression <$> upToNextMinorOf pkgDep
+              , toClosedIntervalExpression <$> closedInterval pkgDep
+              , toRhsHalfOpenIntervalExpression <$> rhsHalfOpenInterval pkgDep
+              ]
+      , dependencyLocations = []
+      , dependencyEnvironments = []
+      , dependencyTags = Map.empty
+      }
+  where
+    -- from constraint is equivalent to upToNextMajor
+    -- Reference: https://github.com/apple/swift-package-manager/blob/main/Documentation/PackageDescription.md#methods-3
+    toFromExpression :: Text -> Text
+    toFromExpression = toUpToNextMajorExpression
+
+    -- Fetcher accepts ^ operator, to perform
+    -- upToNext major constraint validation.
+    toUpToNextMajorExpression :: Text -> Text
+    toUpToNextMajorExpression v = "^" <> v
+
+    -- Fetcher accepts ~ operator, to perform
+    -- upToNext minor constraint validation.
+    toUpToNextMinorExpression :: Text -> Text
+    toUpToNextMinorExpression v = "~" <> v
+
+    toClosedIntervalExpression :: (Text, Text) -> Text
+    toClosedIntervalExpression (lhs, rhs) = ">=" <> lhs <> " " <> "<=" <> rhs
+
+    toRhsHalfOpenIntervalExpression :: (Text, Text) -> Text
+    toRhsHalfOpenIntervalExpression (lhs, rhs) = ">=" <> lhs <> " " <> "<" <> rhs
+
+    depType :: DepType
+    depType =
+      if isJust $ asum [branchOf pkgDep, revisionOf pkgDep, exactOf pkgDep]
+        then GitType
+        else SwiftType
+toDependency (PathSource _) = Nothing
```

Comment:
[nit]: Keep smaller pattern matches closer to the type declaration, so that it's easier to see that they belong together.